# This file is part of HiGFXback

# requires
REQUIRES="make-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure
sh ./Configure --disable-static --soversion=0 --prefix=

# build
make
make libudev.pc

# install
install -d $DESTDIR/include
install -m 644 udev.h $DESTDIR/include/libudev.h
install -d $DESTDIR/lib
install libudev.so.0 $DESTDIR/lib
ln -sf libudev.so.0 $DESTDIR/lib/libudev.so
install -d $DESTDIR/lib/pkgconfig
install -m 644 libudev.pc $DESTDIR/lib/pkgconfig

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/libudev-zero-build.pc << EOF
Name: libudev-zero
Version: 1.0.3
Description: Daemonless Userspace Device management library
Requires: $REQUIRES

devel=\\
/include/libudev.h \\
/lib/libudev.so \\
/lib/pkgconfig/libudev.pc

exec=\\
/lib/libudev.so.0
EOF
ln -sf libudev-zero-build.pc $DESTDIR/share/pkgconfig/libudev-build.pc

exit

# patch
--- libudev-zero-1.0.3.orig/Configure
+++ libudev-zero-1.0.3/Configure
@@ -0,0 +1,21 @@
+#!/bin/sh
+
+PREFIX=/usr/local
+
+SOVERSION=1
+
+while [ $# != 0 ]; do
+  case $1 in
+    --prefix=*) PREFIX=${1#*=};;
+    --soversion=*) SOVERSION=${1#*=};;
+    --disable-static) NOSTATIC=yes;;
+    *) echo "Unknown option: '$1'"; exit 1;;
+  esac
+  shift
+done
+
+sed -i "/^PREFIX/s|=.*|= $PREFIX|" Makefile
+
+sed -i "/^SOVERSION/s|=.*|= $SOVERSION|" Makefile
+
+sed -i "/^NOSTATIC/s|=.*|= $NOSTATIC|" Makefile
--- libudev-zero-1.0.3.orig/Makefile
+++ libudev-zero-1.0.3/Makefile
@@ -1,14 +1,21 @@
 # SPDX-License-Identifier: ISC
 .POSIX:
 
+NOSTATIC ?= 
+
 PREFIX = /usr/local
+SOVERSION = 1
 LIBDIR = ${PREFIX}/lib
 INCLUDEDIR = ${PREFIX}/include
 PKGCONFIGDIR = ${LIBDIR}/pkgconfig
-XCFLAGS = ${CPPFLAGS} ${CFLAGS} -std=c99 -fPIC -pthread -D_XOPEN_SOURCE=700 \
-		  -Wall -Wextra -Wpedantic -Wmissing-prototypes -Wstrict-prototypes \
+XCFLAGS = ${CPPFLAGS} ${CFLAGS} -std=c99 -pthread -D_XOPEN_SOURCE=700 \
+		  -Wall -Wextra -pedantic -Wmissing-prototypes -Wstrict-prototypes \
 		  -Wno-unused-parameter
-XLDFLAGS = ${LDFLAGS} -shared -Wl,-soname,libudev.so.1
+ifneq "$(NOSTATIC)" ""
+XCFLAGS += -fPIC
+endif
+
+XLDFLAGS = ${LDFLAGS} -shared -Wl,-soname,libudev.so.${SOVERSION}
 XARFLAGS = -rc
 AR = ar
 
@@ -19,7 +26,11 @@
 	  udev_monitor.o \
 	  udev_enumerate.o
 
-all: libudev.so.1 libudev.a
+ifneq "$(NOSTATIC)" ""
+all: libudev.so.${SOVERSION}
+else
+all: libudev.a
+endif
 
 .c.o:
 	${CC} ${XCFLAGS} -c -o $@ $<
@@ -27,7 +38,7 @@
 libudev.a: ${OBJ}
 	${AR} ${XARFLAGS} $@ ${OBJ}
 
-libudev.so.1: ${OBJ}
+libudev.so.${SOVERSION}: ${OBJ}
 	${CC} ${XCFLAGS} -o $@ ${OBJ} ${XLDFLAGS}
 
 libudev.pc: libudev.pc.in
@@ -54,26 +65,30 @@
 	mkdir -p ${DESTDIR}${PKGCONFIGDIR}
 	cp -f libudev.pc ${DESTDIR}${PKGCONFIGDIR}/libudev.pc
 
-install-shared: libudev.so.1 install-headers install-pkgconfig
+install-shared: libudev.so.${SOVERSION} install-headers install-pkgconfig
 	mkdir -p ${DESTDIR}${LIBDIR}
-	cp -f libudev.so.1 ${DESTDIR}${LIBDIR}/libudev.so.1
-	ln -fs libudev.so.1 ${DESTDIR}${LIBDIR}/libudev.so
+	cp -f libudev.so.${SOVERSION} ${DESTDIR}${LIBDIR}/libudev.so.${SOVERSION}
+	ln -fs libudev.so.${SOVERSION} ${DESTDIR}${LIBDIR}/libudev.so
 
 install-static: libudev.a install-headers install-pkgconfig
 	mkdir -p ${DESTDIR}${LIBDIR}
 	cp -f libudev.a ${DESTDIR}${LIBDIR}/libudev.a
 
-install: install-shared install-static
+ifneq "$(NOSTATIC)" ""
+install: install-shared
+else
+install: install-static
+endif
 
 uninstall:
 	rm -f ${DESTDIR}${LIBDIR}/libudev.a \
           ${DESTDIR}${LIBDIR}/libudev.so \
-          ${DESTDIR}${LIBDIR}/libudev.so.1 \
+          ${DESTDIR}${LIBDIR}/libudev.so.${SOVERSION} \
           ${DESTDIR}${PKGCONFIGDIR}/libudev.pc \
           ${DESTDIR}${INCLUDEDIR}/libudev.h
 
 clean:
-	rm -f libudev.so.1 libudev.a libudev.pc ${OBJ}
+	rm -f libudev.so.${SOVERSION} libudev.a libudev.pc ${OBJ}
 
 .PHONY: all clean install-headers install-pkgconfig install-shared \
 	install-static install uninstall
--- libudev-zero-1.0.3.orig/udev_device.c
+++ libudev-zero-1.0.3/udev_device.c
@@ -456,13 +456,7 @@
         udev_list_entry_add(&udev_device->properties, "ID_INPUT_SWITCH", "1", 0);
     }
 
-    if (test_bit(ev_bits, EV_REL)) {
-        if (test_bit(rel_bits, REL_Y) && test_bit(rel_bits, REL_X) &&
-            test_bit(key_bits, BTN_MOUSE)) {
-            udev_list_entry_add(&udev_device->properties, "ID_INPUT_MOUSE", "1", 0);
-        }
-    }
-    else if (test_bit(ev_bits, EV_ABS)) {
+    if (test_bit(ev_bits, EV_ABS)) {
         if (test_bit(key_bits, BTN_SELECT) || test_bit(key_bits, BTN_TR) ||
             test_bit(key_bits, BTN_START) || test_bit(key_bits, BTN_TL)) {
             if (test_bit(key_bits, BTN_TOUCH)) {
@@ -492,6 +486,12 @@
             }
         }
     }
+    else if (test_bit(ev_bits, EV_REL)) {
+        if (test_bit(rel_bits, REL_Y) && test_bit(rel_bits, REL_X) &&
+            test_bit(key_bits, BTN_MOUSE)) {
+            udev_list_entry_add(&udev_device->properties, "ID_INPUT_MOUSE", "1", 0);
+        }
+    }
 
     if (!test_bit(ev_bits, EV_KEY)) {
         return;

# source
https://github.com/illiliti/libudev-zero/archive/1.0.3/libudev-zero-1.0.3.tar.gz
